opm/simulators/wells/StandardWellGeneric.cpp

```diff
@@ -199,192 +199,10 @@ StandardWellGeneric<Scalar>::
-    const double flo_bhp_limit = -flo(frates(controls.bhp_limit));
-    if (flo_samples.back() < flo_bhp_limit) {
-        flo_samples.push_back(flo_bhp_limit);
-    }
-    for (double& x : flo_samples) {
-        x = -x;
-    }
-
-    // Find bhp values for inflow relation corresponding to flo samples.
-    std::vector<double> bhp_samples;
-    for (double flo_sample : flo_samples) {
-        if (flo_sample < -flo_bhp_limit) {
-            // We would have to go under the bhp limit to obtain a
-            // flow of this magnitude. We associate all such flows
-            // with simply the bhp limit. The first one
-            // encountered is considered valid, the rest not. They
-            // are therefore skipped.
-            bhp_samples.push_back(controls.bhp_limit);
-            break;
-        }
-        auto eq = [&flo, &frates, flo_sample](double bhp) {
-            return flo(frates(bhp)) - flo_sample;
-        };
-        // TODO: replace hardcoded low/high limits.
-        const double low = 10.0 * unit::barsa;
-        const double high = 600.0 * unit::barsa;
-        const int max_iteration = 50;
-        const double flo_tolerance = 1e-6 * std::fabs(flo_samples.back());
-        int iteration = 0;
-        try {
-            const double solved_bhp = RegulaFalsiBisection<>::
-                solve(eq, low, high, max_iteration, flo_tolerance, iteration);
-            bhp_samples.push_back(solved_bhp);
-        }
-        catch (...) {
-            // Use previous value (or max value if at start) if we failed.
-            bhp_samples.push_back(bhp_samples.empty() ? high : bhp_samples.back());
-            deferred_logger.warning("FAILED_ROBUST_BHP_THP_SOLVE_EXTRACT_SAMPLES",
-                                    "Robust bhp(thp) solve failed extracting bhp values at flo samples for well " + baseif_.name());
-        }
-    }
-
-    // Find bhp values for VFP relation corresponding to flo samples.
-    const int num_samples = bhp_samples.size(); // Note that this can be smaller than flo_samples.size()
-    std::vector<double> fbhp_samples(num_samples);
-    for (int ii = 0; ii < num_samples; ++ii) {
-        fbhp_samples[ii] = fbhp(frates(bhp_samples[ii]));
-    }
-// #define EXTRA_THP_DEBUGGING
-#ifdef EXTRA_THP_DEBUGGING
-    std::string dbgmsg;
-    dbgmsg += "flo: ";
-    for (int ii = 0; ii < num_samples; ++ii) {
-        dbgmsg += "  " + std::to_string(flo_samples[ii]);
-    }
-    dbgmsg += "\nbhp: ";
-    for (int ii = 0; ii < num_samples; ++ii) {
-        dbgmsg += "  " + std::to_string(bhp_samples[ii]);
-    }
-    dbgmsg += "\nfbhp: ";
-    for (int ii = 0; ii < num_samples; ++ii) {
-        dbgmsg += "  " + std::to_string(fbhp_samples[ii]);
-    }
-    OpmLog::debug(dbgmsg);
-#endif // EXTRA_THP_DEBUGGING
-
-    // Look for sign changes for the (fbhp_samples - bhp_samples) piecewise linear curve.
-    // We only look at the valid
-    int sign_change_index = -1;
-    for (int ii = 0; ii < num_samples - 1; ++ii) {
```

Comment:
Are we sure that we want to remove the line-intersection algorithms related to solve the THP controlled wells? @totto82 @atgeirr

Reply:
I am not sure. I did some testing where I added it for both MSW and STW as a last try if everything else failed, but it didn't do much difference.